src/renderers/webgpu/WebGPUBackend.js

```diff
@@ -64,7 +64,7 @@ class WebGPUBackend extends Backend {
 				powerPreference: parameters.powerPreference
 			};
 
-			const adapter = await navigator.gpu.requestAdapter( adapterOptions );
+			const adapter = ( typeof navigator !== 'undefined' ) ? await navigator.gpu.requestAdapter( adapterOptions ) : {};
```

Comment:
@donmccurdy`{}` is intended as a workaround for node but I'm not sure `{}` is ideal here. Do you think this use case is handled appropriately or would you implement it differently?

Reply:
Base on that function code, it should be null, otherwise, next step is exceptional?
"adapter.features"

Reply:
@whatisor are you executing this code in the Node.js environment, or is this a compile-time error from TypeScript or Webpack? Not clear to me from #29916.

I'm not sure that `backend.init( renderer )` should be making efforts to avoid runtime exceptions, if running in an environment where neither WebGL nor WebGPU exists. I would support fixes that avoid build-time errors, and run-time errors in module scope. But if you are calling methods on a renderer in Node.js, a runtime exception is to be expected.

Reply:
> I'm not sure that backend.init( renderer ) should be making efforts to avoid runtime exceptions, if running in an environment where neither WebGL nor WebGPU exists

I agree, I think we should be throwing an error here, it doesn't make much sense to me to proceed with an empty object as an adapter.

Reply:
@donmccurdy it is compile error.
We also need to fix three\examples\jsm\capabilities\WebGPU.js with global and await.
So, we should return null and (exception is next step as current)?

Reply:
> three\examples\jsm\capabilities\WebGPU.js

Capabilities should be optional, it should be possible to use `WebGPURenderer` without global await. https://github.com/mrdoob/three.js/pull/29218

Reply:
can we do same thing with webgl to avoid await?
https://developer.mozilla.org/en-US/docs/Web/API/GPUCanvasContext